src/Symfony/Component/Security/Guard/Tests/Firewall/GuardAuthenticationListenerTest.php

```diff
@@ -184,6 +204,45 @@ public function testHandleCatchesAuthenticationException()
         $listener->handle($this->event);
     }
 
+    /**
+     * @legacy
+     */
+    public function testLegacyInterfaceAvoidSupportsCall()
+    {
+        $authenticatorA = $this->getMock('Symfony\Component\Security\Guard\GuardAuthenticatorInterface');
+        $providerKey = 'my_firewall3';
+
+        $authenticatorA
+            ->expects($this->never())
+            ->method('supports');
```

Comment:
This test fails on php7 with:

> Symfony\Component\Security\Guard\Tests\Firewall\GuardAuthenticationListenerTest::testLegacyInterfaceAvoidSupportsCall
PHPUnit_Framework_MockObject_RuntimeException:
> Trying to configure method "supports" which cannot be configured because it does not exist,
> has not been specified, is final, or is static

I'm not sure you can mock a method that doesn't exist. But, I'm not sure this is necessary at all: if `supports()` *were* accidentally called on this mock, wouldn't it explode with a "Method supports() does not exist" anyways?